src/sonic-yang-models/yang-models/sonic-isis.yang

```diff
@@ -0,0 +1,456 @@
+                        error-message "SPF short delay must only be specified if all other spf parameters are specified";
+                    }
+                    description
+                        "Delay used during SHORT_WAIT state";
+                }
+
+                leaf spf-long-delay {
+                    type uint16 {
+                        range "0..60000";
+                    }
+                    units "msec";
+                    must "(not((not(../spf-long-delay)) and ../spf-init-delay and ../spf-short-delay and ../spf-hold-down and ../spf-time-to-learn))" {
+                        error-message "SPF long delay must only be specified if all other spf parameters are specified";
+                    }
+                    description
+                        "Delay used during LONG_WAIT state";
+                }
+
+                leaf spf-hold-down {
+                    type uint16 {
+                        range "0..60000";
+                    }
+                    units "msec";
+                    must "(not((not(../spf-hold-down)) and ../spf-short-delay and ../spf-long-delay and ../spf-init-delay and ../spf-time-to-learn))" {
+                        error-message "SPF hold down must only be specified if all other spf parameters are specified";
+                    }
+                    description
+                        "Period of time without IGP events before considering IGP stable";
+                }
+
+                leaf spf-time-to-learn {
+                    type uint16 {
+                        range "0..60000";
+                    }
+                    units "msec";
+                    must "(not((not(../spf-time-to-learn)) and ../spf-short-delay and ../spf-long-delay and ../spf-hold-down and ../spf-init-delay))" {
+                        error-message "SPF time-to-learn must only be specified if all other spf parameters are specified";
+                    }
+                    description
+                        "Maximum time needed to learn all of the events related to a
+                        failure";
+                }
+
+                leaf log-adjacency-changes {
+                    type boolean;
+                    default "false";
+                    description
+                        "Log changes to this instance's IS-IS adjacencies.";
+                }
+
+            } // list ISIS_GLOBAL_LIST
+
+        } // container ISIS_GLOBAL
+
+        container ISIS_LEVEL {
+
+            list ISIS_LEVEL_LIST {
+
+                description
+                    "Configuration parameters related to a particular level within the
+                    IS-IS protocol instance";
+
+                key "instance level-number";
+
+                leaf instance {
+                    type string;
+                    description
+                        "The identifier for this instance of ISIS. Area-tag";
+                }
+
+                leaf level-number {
```

Comment:
Openconfig uses dashes within leaf / typedef names. It seems SONiC often uses underscores. Should I standardize based on openconfig or SONiC standards here ?